src/engine.ts

```diff
@@ -288,13 +292,14 @@ export class Engine<A extends string = never, T extends Task<A> = Task<A>> {
    * @param task The current executing task.
    */
   do(task: T): void {
+    const macroString = Macro.load().toString();
```

Comment:
The macroString needs to have a semicolon added if it doesn't already, because Mafia parses 1-step macros without semicolons as a combat script to call. We can just always add a concluding semicolon.